$NetBSD: patch-configure,v 1.4 2020/07/27 16:06:08 manu Exp $

Make FUSE notification optional at configure time
From upstream https://review.gluster.org/24731

NetBSD build fixes
From upstream https://review.gluster.org/24648

--- configure.orig	2020-07-06 03:22:43.222722780 +0200
+++ configure	2020-07-22 02:28:26.017707376 +0200
@@ -932,8 +932,9 @@
 with_tmpfilesdir
 with_ocf
 with_server
 enable_fuse_client
+enable_fuse_notifications
 enable_fusermount
 enable_epoll
 enable_georeplication
 enable_events
@@ -1619,8 +1620,10 @@
                           Disable internal tracking of privileged ports.
   --enable-valgrind       Enable valgrind for resource leak debugging.
   --disable-fuse-client   Do not build the fuse client. NOTE: you cannot mount
                           glusterfs without the client
+  --disable-fuse-notifications
+                          Disable FUSE notifications
   --disable-fusermount    Use system's fusermount
   --disable-epoll         Use poll instead of epoll.
   --disable-georeplication
                           Do not install georeplication components
@@ -14172,8 +14175,23 @@
 if test "x$enable_fuse_client" != "xno"; then
    FUSE_CLIENT_SUBDIR=fuse
    BUILD_FUSE_CLIENT="yes"
 fi
+  
+  
+                          
+# Check whether --enable-fuse-notifications was given.
+if test "${enable_fuse_notifications+set}" = set; then :
+  enableval=$enable_fuse_notifications;
+fi
+  
+                          
+if test "x$enable_fuse_notifications" != "xno"; then :    
+  
+  
+$as_echo "#define HAVE_FUSE_NOTIFICATIONS 1" >>confdefs.h
+  
+fi
 
 
 # end FUSE section
 
@@ -14973,8 +14991,11 @@
     *freebsd*)
     CFLAGS="${CFLAGS} -isystem /usr/local/include"
     ARGP_LDADD=-largp
     ;;
+    *netbsd*)
+    ARGP_LDADD=-largp
+    ;;
 esac
 ac_fn_c_check_header_mongrel "$LINENO" "argp.h" "ac_cv_header_argp_h" "$ac_includes_default"
 if test "x$ac_cv_header_argp_h" = xyes; then :
 
